hazelcast/src/main/java/com/hazelcast/sql/impl/plan/node/io/UnicastSendPlanNode.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.plan.node.io;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.sql.impl.plan.node.PlanNode;
+import com.hazelcast.sql.impl.plan.node.UniInputPlanNode;
+import com.hazelcast.sql.impl.partitioner.RowPartitioner;
+import com.hazelcast.sql.impl.plan.node.PlanNodeVisitor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Node which unicasts data to remote stripes.
+ */
+public class UnicastSendPlanNode extends UniInputPlanNode implements EdgeAwarePlanNode {
+    /** Edge ID. */
+    private int edgeId;
+
+    /** Partition hasher (get partition hash from row). */
+    private RowPartitioner partitioner;
+
+    public UnicastSendPlanNode() {
+        // No-op.
+    }
+
+    public UnicastSendPlanNode(int id, PlanNode upstream, int edgeId, RowPartitioner partitioner) {
+        super(id, upstream);
+
+        this.edgeId = edgeId;
+        this.partitioner = partitioner;
+    }
+
+    public RowPartitioner getPartitioner() {
```

Comment:
The `partitioner` is not used, which indicates an incorrect implementation of the `CreateExecPlanNodeVisitor.onUnicastSendNode`.

Reply:
Assert that SingleValuePartitioner is used https://github.com/hazelcast/hazelcast/pull/18013/commits/f47bac35ee4dbb1570cd92cf5c1a4188135f4194
This part will change when we implement a more generic plan generation.

Reply:
I am concerned that the partitioner is still not used. Why don't we pass it to the `UnicastSendExec`? This leaves PR in a strange state: we introduce partitioners but do not use them. Let's either remove them completely and reintroduce them later, or integrate them properly.